modules/core/src/scenegraph/gltf-instantiator/gltf-instantiator.js

```diff
@@ -173,14 +212,13 @@ export default class GLTFInstantiator {
         gltfTexture.texture.sampler &&
         gltfTexture.texture.sampler.parameters) ||
       {};
-    gltfTexture.texture.source.image.then(image => {
-      const tex = new Texture2D(this.gl, {
-        id: gltfTexture.name || gltfTexture.id,
-        parameters,
-        data: image
-      });
-      model.setUniforms({tex});
+
+    const tex = new Texture2D(this.gl, {
+      id: gltfTexture.name || gltfTexture.id,
+      parameters,
+      data: gltfTexture.texture.source.getImageAsync()
```

Comment:
Maybe a short comment explaining that we are passing in a promise and that this is supported by the texture.